lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,432 @@
+import jwt
+from flask import request, jsonify, make_response
+from flask_restful import Resource, abort
+from flask_bcrypt import check_password_hash
+from email_validator import validate_email, EmailNotValidError
+from configparser import NoOptionError
+from pbench.server.api.resources.models import UserModel, ActiveTokens
+from pbench.server.api.resources.auth import auth, encode_auth_token
+from pbench.server.api.resources.database import Database
+
+
+class RegisterUser(Resource):
+    """
+    Abstracted pbench API for registering a new user
+    """
+
+    def __init__(self, config, logger):
+        self.server_config = config
+        self.logger = logger
+        self.bcrypt_log_rounds = int(
+            self.server_config.get("pbench-server", "bycrypt_log_rounds")
+        )
+        try:
+            self.token_expire_duration = self.server_config.get(
+                "pbench-server", "token_expiration_duration"
+            )
+        except NoOptionError:
+            self.token_expire_duration = None
```

Comment:
Do we really want to silently allow tokens not to expire?

Perhaps a log message here, or perhaps we should always require that setting.

Reply:
Yes I was confused between 2 options, 1. having some default value and put it in the `server-default.config` and if we want to override then we add it to `server.config`. Or we have a strict requirement of having this field in the `server.config` file, how should we do it?

Reply:
If we put it in `pbench-server-defaults.cfg` they only need to worry about it if they want to override the value in `pbench-server.cfg` ... if they don't, we just get our default. And if someone's foolish enough to delete it from our `pbench-server-defaults.cfg` then they get what they deserve: I'd be OK with an error, or with an arbitrary default like 1 second. ;-)

Reply:
Bump: this should be in the standard `pbench-server-default.cfg` and possibly in the `pbench-server.cfg.example` if we want it to be easy to override. And if you want a failsafe fallback should they delete that, I'd be happy to have something "sane" like a day, but `None` is probably not the wisest choice.

Reply:
Fixed